tandemhunter-main/TandemHunter.py

```diff
@@ -0,0 +1,465 @@
+    )
+    required.add_argument(
+        "--intervals",
+        type=str,
+        required=True,
+        help=(
+            "Path to json file containing intervals to compare coverage at."
+            " These regions have to be in the coverage file as is as this "
+            " programme uses pattern matching"
+        ),
+    )
+    parser.add_argument(
+        "--dup-threshold",
+        type=float,
+        default=1.122995,
+        help="Default threshold for dup/amp regions if not in interval file",
+    )
+    parser.add_argument(
+        "-O", "--out_dir", default=".", help="Output write directory"
+    )
+    parser.add_argument(
+        "--out-fname-suffix",
+        default=".cvg_comparison.csv",
+        help="Filename suffix to use when naming results file",
+    )
+    parser.add_argument(
+        "--processes", default=0,
+        help="Number of processes to run in parallel",
+    )
+    parser.add_argument(
+        "--cov-file-pattern",
+        default="coverage.tsv",
+        help="Filename suffix used to identify coverage files in batch mode",
+    )
+    parser.add_argument(
+        "--metric",
+        default="normalized_coverage",
+        help="Metric to used in comparison",
+    )
+    parser.add_argument("--version",
+                        action="version", version="%(prog)s 1.0.0")
+
+    return parser.parse_args(args)
+
+
+class TDHunter(object):
+    def __init__(
+        self,
+        file=False,
+        batch=False,
+        intervals=False,
+        dup_threshold=1.122995,
+        out_dir=".",
+        out_fname_suffix=".cvg_comparison.csv",
+        processes=0,
+        cov_file_pattern=".qc.coverage.txt",
+        column_dtypes=COV_COLUMN_DTYPES,
+        metric="normalized_coverage",
+    ):
+        """ Object containg methods that compare normalised coverage
+        at two given intervals.
+        :param: file: Path to coverage file
+        :param: batch: Path to folder with a batch of coverage files
+        :param: intervals: Path to file with intervals to compare coverage at
+        :param: dup_threshold: Threshold used to identify dup/amp regions
+        :param: out_dir: Output write directory
+        :param: out_fname_suffix: Filename suffix for naming results file
+        :param: processes: Maximum number of processes to run (default is cpu
+                            count)
+        :param: cov_file_pattern: Filename suffix to identify coverage files
+                                    when in batch mode
+        :param: column_dtypes: Dictionary with column name and datatype for
+                            the given coverage file type
+        :param: metric: Metric to used in comparison
+        :returns: Writes a coverage comparison between two intervals to a text
+                    file on per-sample basis
+        """
+
+        # store arguments
+        try:
+            self.file = os.path.abspath(file)
+        except Exception:

```

Comment:
Better to be more specific with the Exception, if possible, by naming the ones you're expecting - otherwise you hoover up all possible problems, like keyboard interrupts, not just ones caused by there being no file. You'll need to investigate a bit but when I tried it locally just now, running os.path.abspath() with no input gives TypeError: abspath() missing 1 required positional argument: 'path'. So, you need 'except TypeError' . You might also need OSError as this is from the 'os' package. Together this can be written: 'except (TypeError, OSError):'